src/data/complex/module.lean

```diff
@@ -78,6 +78,17 @@ instance [comm_semiring R] [algebra R ℝ] : algebra R ℂ :=
   commutes' := λ r ⟨xr, xi⟩, by ext; simp [smul_re, smul_im, algebra.commutes],
   ..complex.of_real.comp (algebra_map R ℝ) }
 
+/-- Two `ℝ`-algebra homomorphisms from ℂ are equal if they agree on `complex.I`. -/
+@[ext]
+lemma alg_hom_ext {A : Type*} [semiring A] [algebra ℝ A] ⦃f g : ℂ →ₐ[ℝ] A⦄ (h : f I = g I) :
+  f = g :=
+begin
+  ext ⟨x, y⟩,
+  have : (coe : ℝ → ℂ) = algebra_map ℝ ℂ := rfl,
+  rw [mk_eq_add_mul_I, ←smul_coe, this],
+  simp only [alg_hom.map_add, alg_hom.commutes, alg_hom.map_smul, h],
+end
```

Comment:
If we don't go with the `power_basis` approach, then we can put `alg_hom.commutes` in `simp` normal form instead:
```suggestion
@[simp] lemma _root_.alg_hom.map_real_cast {A : Type*} [semiring A] [algebra ℝ A]
  (f : ℂ →ₐ[ℝ] A) (x : ℝ) :
  f x = algebra_map ℝ A x :=
f.commutes x

/-- Two algebra homomorphisms from ℂ are equal if they agree on `complex.I`. -/
@[ext]
lemma alg_hom_ext {A : Type*} [semiring A] [algebra ℝ A] ⦃f g : ℂ →ₐ[ℝ] A⦄ (h : f I = g I) :
  f = g :=
begin
  ext ⟨x, y⟩,
  simp only [mk_eq_add_mul_I, alg_hom.map_add, alg_hom.map_real_cast, alg_hom.map_mul, h]
end
```

Reply:
I don't think `alg_hom.map_real_cast` is a good name - it's about `complex.has_coe`, not some generic "real cast".

Reply:
Is `alg_hom.map_coe_real_complex` ok with you?

Reply:
Sounds good as well. I wanted to imitate the family `ring_hom.{nat,int,rat}_cast`, but that is not completely accurate here.

Reply:
Ok, added, along with a comment explaining why it exists

Reply:
We probably could define a `real.cast` for any `[algebra ℝ A]`, but at that point we should probably stop ourselves and decide whether we should just make `coe` the simp-normal form for all `algebra_map`s. The whole reason we need the lemma you suggest is because half of mathlib says "coe is the simp-normal form", and the other half says "algebra_map is".

Reply:
Exactly my thoughts :)